### ISSUE TITLE
Chart doesn't take up full width after redraw

### ISSUE THREAD

**Issue by mshcheng** [PhilJay/MPAndroidChart#2646]
Before i ask my question, let me first say that this is the most powerful and feature-rich charting library I have used on the Android platform.  The documentation is great, and i can find lots of resources on stackoverflow as well.

But i recently come across one issue that i don't know it is a bug or i just haven't done the right thing.  I have setup this chart object to occupy full width of the layout/screen.  On the initial draw with dataset A, the chart appears fine, taking up the full width.  Note that the left Y-axis labels are fairly long in this case.   

![before](https://cloud.githubusercontent.com/assets/11400565/21580376/30f27f64-d02c-11e6-982b-58dc758b85c6.png)

When i redraw the chart with different dataset B, with shorter Y-axis labels, the chart doesn't seem to take the full width of the screen (ie. leaving a wide blank space on the left side).

![after](https://cloud.githubusercontent.com/assets/11400565/21580379/37fd5d4c-d02c-11e6-87c6-71dd077067e0.png)

But if at the very start, i select dataset B (without selecting dataset A first), the chart is drawn properly without leaving a wide blank space on the left.

![proper](https://cloud.githubusercontent.com/assets/11400565/21580396/0c160b2e-d02d-11e6-8642-76fb1ab4c595.png)

Have i missed something here ? any help is appreciated.

**Comment by patrick-iv** [comment]
Hi, are you calling `chart.invalidate()` after changing the data set?

**Comment by mshcheng** [comment]
Yup i did.   Below is the code that i use to draw the chart with.  I basically only change the currentEntries arraylist when i switch between dataset A and B.

```
currentSet = new LineDataSet(currentEntries, "current_carbon");
currentSet.setColor(ContextCompat.getColor(chart.getActivity(), R.color.current_carbon_line));
currentSet.setLineWidth(1f);
currentSet.setCircleColor(ContextCompat.getColor(chart.getActivity(), R.color.current_carbon_line));
currentSet.setCircleRadius(3f);
currentSet.setDrawCircleHole(true);
currentSet.setDrawValues(false);
dataSets.add(currentSet);
LineData lData = new LineData(dataSets);

XAxis xAxis = chart.getXAxis();
xAxis.setValueFormatter(new MealAxisValueFormatter(chart.getActivity(), chart));
xAxis.setGranularity(1f);
xAxis.setLabelRotationAngle(90);
xAxis.setAxisMinValue(0f);
xAxis.setAxisMaxValue((float) (numOfDays*numOfMeals)-1);
xAxis.setPosition(XAxis.XAxisPosition.BOTTOM);
xAxis.setLabelCount(numOfDays*numOfMeals);
chart.getAxisLeft().setAxisMinValue(0f);
chart.getAxisRight().setEnabled(false);
chart.setDrawBorders(true);

chart.clearValues();
chart.setData(lData);
chart.animateX(1000);
chart.invalidate();
chart.fitScreen();
```

**Comment by patrick-iv** [comment]
Does it work if you add `chart.notifyDataSetChanged()`?

**Comment by mshcheng** [comment]
Tried that just now, still the same.

**Comment by tamslinn** [comment]
Did work out an answer to this? I am having the same problem, when I redraw with shorter series, so the x axis is reduced in length. Redrawing longer is fine.

**Comment by tamslinn** [comment]
I got mine to work by recreating the LineData object rather than just amending the values in the LineDataSet

**Comment by jeremy96yg** [comment]
Hi, I just got that problem, just call calcMinMax() to compute again the min and max of the axis on your dataset like that : `chart.lineData.dataSets.first().calcMinMax()`

**Comment by bachpx2** [comment]
@tamslinn thanks for the suggestion, I got mine to work by doing that as well

**Comment by ak4550** [comment]
anyone found the solution?
